packages/concerto-core/lib/decoratormanager.js

```diff
@@ -200,26 +235,87 @@ class DecoratorManager {
     }
 
     /**
-     * Applies all the decorator commands from the DecoratorCommandSet
-     * to the ModelManager.
+     * Add decorator commands set with index object to the coresponding target map
+     * @param {*} targetMap the target map to add the command to
+     * @param {targetKey} targetKey the target key to add the command to
+     * @param {DcsIndexWrapper} dcsWithIndex the command to add
+     * @private
+     */
+    static addDcsWithIndexToMap(targetMap, targetKey, dcsWithIndex) {
+        const targetCommands = targetMap.get(targetKey);
+        if (targetCommands) {
+            targetCommands.push(dcsWithIndex);
+        } else {
+            targetMap.set(targetKey, [dcsWithIndex]);
+        }
+    }
+
+
+    /**
+     * Creates five different maps to index decorator command sets by target type and returns them
+     * @param {*} decoratorCommandSet the DecoratorCommandSet object
+     * @returns {Object} a new model manager with the decorations applied
+     * @private
+     */
+    static getDecoratorMaps(decoratorCommandSet) {
+        const namespaceCommandsMap = new Map();
+        const declarationCommandsMap = new Map();
+        const propertyCommandsMap = new Map();
+        const mapElementCommandsMap = new Map();
+        const typeCommandsMap = new Map();
+
+        decoratorCommandSet.commands.map((decoratorCommand, index) => {
+            const dcsWithIndex = new DcsIndexWrapper(decoratorCommand, index);
+            switch (true) {
```

Comment:
Is this  switch supposed to be exhaustive? If yes, add a default case and throw an exception.

Reply:
I was under the assumption that if we pass a invalid dcs, it will only get caught if validation is enabled. In the case of a malformed decorator, if we pass (in this case target is undefined, falsy, null or does not match with any of the switch statements) that decorator won't get applied.

This is the older behavior: https://replit.com/@sanketshevkar/Decorator-malformed-dcs#index.js
This differs a bit, I think is was a bug as well. shouldn't `target: false` be invalid dcs, but its passing the syntax validation. In either case it just applies the decorator on top of all the declarations.